### PR TITLE
feat(cli): don't check permissions for statically analyzable dynamic imports

### LINKED ISSUE #17658: Dynamic imports require net access even when they are unused
```js
if (false) {
	await import("https://deno.land/std@0.176.0/node/module.ts")
}
```
```
> % deno run foo.js
⚠️  ┌ Deno requests net access to "deno.land".
   ├ Requested by `import()` API
   ├ Run again with --allow-net to bypass this prompt.
   └ Allow? [y/n] (y = yes, allow; n = no, deny) > 
```

I feel like it's only supposed to request access the moment the `await import()` is called, or am I missing something?

For the moment a workaround is to keep the import specifier in a separate variable:
```js
if (false) {
	const str = "https://deno.land/std@0.176.0/node/module.ts";
	const mod = await import(str);
}
```
But this removes type safety and turns `mod` into `any`.

### LINKED ISSUE #17697: Do not permission prompt for statically analyzable dynamic imports
Statically analyzable dynamic imports are part of the graph on startup and are inspectable in the code similarly to any regular import declaration. They also show up in `deno info`. Perhaps we should not bother requiring `--allow-net` when they load?

Related: #14877

### PR DESCRIPTION
**Pull request by nayeemrmn**
Closes #17697. Hoping to get this in a minor release, see https://github.com/denoland/deno/issues/17697#issuecomment-1486208485.
Closes #17658.

### CODE REVIEW

**Review by dsherret — APPROVED** [review]
LGTM. I think this makes sense given these imports show up in the output of `deno info` and are part of the module graph. Someone else should review this one in addition to me before we merge it though.

**Review by aapoalas — APPROVED** [review]
I am also in favour.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**
